### PR TITLE
Various small suggestions

### PR DESCRIPTION
**Pull request by ghost**
Here are some small suggestions that I hope (slightly) improve the look of the code, they shouldn't change the code functionality at all.
I've only taken a few programming classes (none of them for python) so who knows if these are actually good suggestions.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -10,36 +10,30 @@
 port = '8989' #YOUR SONARR PORT
 #END
 
-ignore_default_episdode_name = True # Ignore files where the Episode uses default episode naming?
-scrub__and__strings = True # scrub 'and' from file name strings. Fixes & vs 'and' mismatches
+IGNORE_DEFAULT_EPISODE_NAME = True # Ignore files where the Episode uses default episode naming?
```
Comment (ghost) [inline main.py:13]: All caps differentiates between static globals and normal variables.

File: main.py [hunk 2]
```diff
@@ -10,36 +10,30 @@
...
 z=0 # global counter
 
 
 def default_episode(title):
     """Check whether the episode title has generic naming applied."""
-    if title[:7] == "Episode" and str.isnumeric(title[8:]):
-        return True
-    
-    return False
+    return (title[:7] == "Episode" and str.isnumeric(title[8:]))
```
Comment (ghost) [inline main.py:24]: The conditional already does the work of the if and return statements so why not just return the conditional. (this was a big pet peeve of one of my CS professors)

File: main.py [hunk 3]
```diff
@@ -10,36 +10,30 @@
...
     """Check whether the file name covers multiple episdoes."""
-    if bool(re.search('(s\d{1,4}e\d{1,4}-\d{1,4})+',file_string)):
-        return True
-
-    return False
+    return bool(re.search('(s\d{1,4}e\d{1,4}-\d{1,4})+',file_string))
 
 
-def process_season(series_id):
+def process_season(series_id, series_name):
```
Comment (ghost) [inline main.py:32]: passing `series_name` allows you to not have to use a global.

File: main.py [hunk 4]
```diff
@@ -79,23 +73,23 @@ def process_season(series_id):
...
                     # But this only gets applied where we already know it's a multi episode file, so shouldn't cause too many issues
 
-                if default_episode(title) and ignore_default_episdode_name:
+                if default_episode(title) and IGNORE_DEFAULT_EPISODE_NAME:
                     # This function tests is the episode title is just 'default' naming. eg 'Episode 1'
                     # In these cases, the user might not want/need 'Episode 1' in the file name.
                     # What I'd like to do is then test the S01E01 matches, but for now, we'll just test for it with the option to skip these files
 
-                    1==1
+                    pass
```
Comment (ghost) [inline main.py:88]: `pass` is meant for these exact situations where you don't actually want to do anything. Its a no-op/null operation.

File: main.py [hunk 5]
```diff
@@ -108,11 +102,10 @@ def process_season(series_id):
                         if is_multi_episode(str.lower(file)):
                             text_file.write("MULTIFILE: ")
                         text_file.write("Mismatch Found: %s" % title + " | " + file + "\n")
-                    z=z+1
+                    z+=1
```
Comment (ghost) [inline main.py:101]: personal preference. Saves a whole character!

**Comment by ghost** [comment]
Oh! two more things.

    #UPDATE THESE THREE VALUES!!
    apikey = '000000' #YOUR API KEY
    url = 'http://192.168.1.208' #YOUR SONARR URL
    port = '8989' #YOUR SONARR PORT
    #END

Personally, I would change this and either put it in a separate settings file that you load at runtime (this makes it so the user doesn't have to edit the actual program file) or pass the arguments through the commandline (very versatile and quick from the user perspective but thats not really needed in this case since these options shouldn't really be changing often) or both. This also stops you from accidentilly uploading secrets to github. I almost just did this with my API key (EDIT:  D'OH! never mind I actually uploaded my API key). Just an API key for sonarr wouldn't be that useful but I think its good practice to keep user specific stuff out of the code.

     global z
I'm not a fan of globals in general (big nono for my professors), if this was C++ (the language I was actually taught) I would just pass it by reference but it seems python doesn't really work that way. I'm looking in to the proper python way of doing this.

File: main.py (hunk 1 above)
Comment (ghost) [reply main.py:13]: I don't know how standard this is, I've seen it a few places though. My editor (vscode) highlights them differently which makes them easy to differentiate.

File: main.py [hunk 6]
```diff
@@ -124,7 +116,7 @@ def main():
     for element in json_series:
         series_id = element['id']
         series_name = element['title']
-        process_season(series_id)
+        z += process_season(series_id, series_name)
```
Comment (ghost) [inline main.py:119]: Globals actually don't seem as bad in python as in C++ (I forget why globals are even bad in the first place) but if I were to do this without globals I would do it something like this. Keep a local counter for each call of `process_season` and then add it to the counter total after `process_season` returns.

File: main.py (hunk 4 above)
Comment (extrobe) [reply main.py:88]: Thanks for the tip on using pass - exactly what I needed!

File: main.py (hunk 2 above)
Comment (extrobe) [reply main.py:24]: makes sense, thanks!

File: main.py (hunk 1 above)
Comment (extrobe) [reply main.py:13]: Nice, I'll give this approach a go

**Review by extrobe — APPROVED** [review]
Thanks for the tips & changes - teaching me a few new bits (like you, Python isn't my main strength!

**Comment by extrobe** [comment]
> Oh! two more things.
> 
> ```
> #UPDATE THESE THREE VALUES!!
> apikey = '000000' #YOUR API KEY
> url = 'http://192.168.1.208' #YOUR SONARR URL
> port = '8989' #YOUR SONARR PORT
> #END
> ```
> 
> Personally, I would change this and either put it in a separate settings file that you load at runtime (this makes it so the user doesn't have to edit the actual program file) or pass the arguments through the commandline (very versatile and quick from the user perspective but thats not really needed in this case since these options shouldn't really be changing often) or both. This also stops you from accidentilly uploading secrets to github. I almost just did this with my API key (EDIT: D'OH! never mind I actually uploaded my API key). Just an API key for sonarr wouldn't be that useful but I think its good practice to keep user specific stuff out of the code.
> 
> ```
>  global z
> ```
> 
> I'm not a fan of globals in general (big nono for my professors), if this was C++ (the language I was actually taught) I would just pass it by reference but it seems python doesn't really work that way. I'm looking in to the proper python way of doing this.

Yeah, a colleague said the same thing, and given me some pointers for that. So that will probably be next on my list! Thanks for taking a look and making the other suggestions!

**Pull request merged**